yt_dlp/extractor/youtube.py

```diff
@@ -2212,83 +2219,33 @@ def extract_thread(contents):
             response = self._extract_response(
                 item_id=None, query=continuation,
                 ep='next', ytcfg=ytcfg, headers=headers, note=note_prefix,
-                check_get_keys=('onResponseReceivedEndpoints', 'continuationContents'))
-            if not response:
-                break
-            visitor_data = try_get(
-                response,
-                lambda x: x['responseContext']['webResponseContextExtensionData']['ytConfigData']['visitorData'],
-                compat_str) or visitor_data
+                check_get_keys='onResponseReceivedEndpoints')
 
-            continuation_contents = dict_get(response, ('onResponseReceivedEndpoints', 'continuationContents'))
+            continuation_contents = traverse_obj(
+                response, 'onResponseReceivedEndpoints', expected_type=list, default=[])
 
             continuation = None
-            if isinstance(continuation_contents, list):
-                for continuation_section in continuation_contents:
-                    if not isinstance(continuation_section, dict):
-                        continue
-                    continuation_items = try_get(
-                        continuation_section,
-                        (lambda x: x['reloadContinuationItemsCommand']['continuationItems'],
-                         lambda x: x['appendContinuationItemsAction']['continuationItems']),
-                        list) or []
-                    if is_first_continuation:
-                        continuation = extract_header(continuation_items)
-                        is_first_continuation = False
-                        if continuation:
-                            break
-                        continue
-                    count = 0
-                    for count, entry in enumerate(extract_thread(continuation_items)):
-                        yield entry
-                    continuation = self._extract_continuation({'contents': continuation_items})
+            for continuation_section in continuation_contents:
+                continuation_items = try_get(
+                    continuation_section,
+                    (lambda x: x['reloadContinuationItemsCommand']['continuationItems'],
+                     lambda x: x['appendContinuationItemsAction']['continuationItems']),
+                    list) or []
```

Comment:
pretty sure this is same as:
```suggestion
                continuation_items = get_first(
                    continuation_section,
                    ('reloadContinuationItemsCommand', 'appendContinuationItemsAction'), 'continuationItems',
                    expected_type=list) or []
```

Reply:
hmm not quite

```suggestion
                continuation_items = get_first(
                    continuation_section,
                    [[('reloadContinuationItemsCommand', 'appendContinuationItemsAction'), 'continuationItems']],
                    expected_type=list) or []
```
but this seems to work?

Reply:
hm.. I though `get_first` takes `*args`, but seems not
```py
def get_first(obj, keys, **kwargs):
    return traverse_obj(obj, (..., *variadic(keys)), **kwargs, get_all=False)
```
So it should be `(('reloadContinuationItemsCommand', 'appendContinuationItemsAction'), 'continuationItems')`. Are you sure there needs to be 2 levels of nesting?

Reply:
`(('reloadContinuationItemsCommand', 'appendContinuationItemsAction'), 'continuationItems')` doesn't work either. 

But with two levels of nesting it seems to work

Reply:
sorry, get_first isn't suitable here. I was thinking directly in terms of `traverse_obj`
```
>>> from yt_dlp.utils import traverse_obj
>>> d = {'x': {'y': 'z'}}
>>> traverse_obj(d, (('x', 'a'), 'y'), get_all=False)
'z'
```
So:
```suggestion
                continuation_items = traverse_obj(
                    continuation_section,
                    (('reloadContinuationItemsCommand', appendContinuationItemsAction'), 'continuationItems'),
                    get_all=False, expected_type=list) or []
```